### PR TITLE
Use circulation rules to drive patron-specific item request types

### LINKED ISSUE #2097: Use patron-specific circulation rules (or FOLIO APIs?) to figure out whether to place a Hold or a Recall
E.g. https://requests.stanford.edu/hold_recalls/379379/status should have been placed as a Hold, not a Recall, because the patron's group doesn't allow Recalls.

It looks like we initially relied on FOLIO's `mod-patron` to assign the proper request type, but when we switched to using `mod-circulation` (for reasons... proxy users, perhaps?) we continued to use the generic circulation rule, not a patron-specific rule.

Here's where we're sending the request type:
https://github.com/sul-dlss/sul-requests/blob/main/app/jobs/submit_folio_request_job.rb#L92C23-L92C45

And the implementation of best_request_type: 
https://github.com/sul-dlss/sul-requests/blob/main/app/models/folio/item.rb#L179

We'll need a patron-aware version that evaluates the request type for the specific patron (and presumably opening https://github.com/sul-dlss/sul-requests/blob/main/app/services/folio/circulation_rules/policy_service.rb#L6 up to accept a specific patron group). 

If, when evaluating the specific patron group permissions, they are ineligible to hold or recall an item (but one of the "standard" patron groups can do one or the other), I think we should fall back on a regular `Hold` request until we can provide more specific messaging to the user.

### PR DESCRIPTION
**Pull request by dnoneill**
closes #2097 

I used the Fee Borrower user and http://localhost:3000/hold_recalls/new?barcode=36105232863154&item_id=14720993&origin=GREEN&origin_location=GRE-STACKS for testing.

### CODE REVIEW

File: app/models/folio/location.rb [hunk 1]
```diff
@@ -7,8 +7,8 @@ module Folio
     def self.from_hash(dyn)
       new(
         id: dyn.fetch('id'),
-        campus: (Campus.new(**dyn.fetch('campus')) if dyn['campus']),
-        library: (Library.new(**dyn.fetch('library').symbolize_keys) if dyn['library']),
+        campus: (Campus.new(**dyn.fetch('campus').slice('id', 'code')) if dyn['campus']),
+        library: (Library.new(**dyn.fetch('library').slice('id', 'code', 'name').symbolize_keys) if dyn['library']),
```
Comment (thatbudakguy) [inline app/models/folio/location.rb:11]: does `Campus` not need symbol keys?

**Review by thatbudakguy — APPROVED** [review]

File: app/models/folio/location.rb (hunk 1 above)
Comment (dnoneill) [reply app/models/folio/location.rb:11]: @cbeer do you know? @thatbudakguy Hasn't caused any problems in testing.

File: app/services/folio/types.rb [hunk 2]
```diff
@@ -84,15 +84,15 @@ def locations
...
-
     def get_type(type)
       raise "Unknown type #{type}" unless types_of_interest.include?(type.to_s)
 
       file = cache_dir.join("#{type}.json")
       JSON.parse(file.read) if file.exist?
     end
 
+    private
+
```
[the remark this answers is not in the document]
Comment (cbeer) [reply app/services/folio/types.rb:95]: (We're using it down in the tests to load up actual type data)

[the remark this answers is not in the document]
Comment (thatbudakguy) [reply app/services/folio/types.rb:95]: Thanks for clarifying!

**Pull request merged**